pkg/qemu/qemu_driver.go

```diff
@@ -411,25 +430,6 @@ func (l *LimaQemuDriver) GuestAgentConn(ctx context.Context) (net.Conn, error) {
 	return dialContext, err
 }
 
-func (l *LimaQemuDriver) CheckBinarySignature(_ context.Context, arch string) error {
```

Comment:
For readability you can just keep this function with privatization (`Check...` -> `check...`), and call it from `Validate()`